sc-poc/managers/configman/adapter/adapter.go

```diff
@@ -12,4 +15,16 @@ type Adapter interface {
 
 	// GetRawConfig returns the config in bytes.
 	GetRawConfig() (common.ConfigType, error)
+
+	// List returns all registered sources of a specific source type
+	List(schema.GroupVersionResource) (*unstructured.UnstructuredList, error)
+
+	// Get returns a registered source
+	Get(schema.GroupVersionResource, string) (*unstructured.Unstructured, error)
+
+	// Apply creates/updates a source
+	Apply(schema.GroupVersionResource, *unstructured.Unstructured) error
```

Comment:
We don't need `schema.GroupVersionResource` for apply. You can generate it from the object to be applied. We can still choose to keep it if you want.